# EDA #
The script has gathered 100 reviews per user for almost 300 users. Time to make sure all the data is present

In [22]:
import pandas as pd

In [31]:
# File path to the consolidated data
path = '../data/interim/consolidated_data.csv'

In [23]:
df = pd.read_csv(path)
df.head()

title  author  avg_rating  \
0  Harry Potter and the Sorcerer's Stone (Harry P...     NaN        4.47   
1            The Hunger Games (The Hunger Games, #1)     NaN        4.33   
2                              To Kill a Mockingbird     NaN        4.26   
3                                   The Great Gatsby     NaN        3.93   
4                                               1984     NaN        4.19   

   user_rating                                          cover_url  user_id  
0            5  https://i.gr-assets.com/images/S/compressed.ph...        1  
1            5  https://i.gr-assets.com/images/S/compressed.ph...        1  
2            5  https://i.gr-assets.com/images/S/compressed.ph...        1  
3            4  https://i.gr-assets.com/images/S/compressed.ph...        1  
4            5  https://i.gr-assets.com/images/S/compressed.ph...        1

In [24]:
df.shape

(29530, 6)

In [26]:
df['title'].nunique()

6334

In [34]:
df['cover_url'].nunique()

9872

In [28]:
df.isna().sum()

title              0
author         29530
avg_rating         0
user_rating        0
cover_url          0
user_id            0
dtype: int64

### First Impressions ###
There are 29,530 reviews here, but only 6,334 unique books. This isn't a huge surprise, since I sorted each user's "read" shelf by most-reviewed and scraped the top 100. I'm actually glad to see this much variety. This should mean that I have legitimate recommendation power but that I don't recommend very obscure or unpopular books. There are more unique cover URLs due to different editions existing for many of the books.

The script also failed to gather the "author" column for some reason. This information would be nice to have but ultimately isn't worth the time to re-run the script. If this project were going into production, I'd gather far more than title, author, and rating anyways. I'll drop the author column for now.

In [33]:
try:
    df = df.drop('author', axis = 1)
except:
    pass
df.head()

title  avg_rating  user_rating  \
0  Harry Potter and the Sorcerer's Stone (Harry P...        4.47            5   
1            The Hunger Games (The Hunger Games, #1)        4.33            5   
2                              To Kill a Mockingbird        4.26            5   
3                                   The Great Gatsby        3.93            4   
4                                               1984        4.19            5   

                                           cover_url  user_id  
0  https://i.gr-assets.com/images/S/compressed.ph...        1  
1  https://i.gr-assets.com/images/S/compressed.ph...        1  
2  https://i.gr-assets.com/images/S/compressed.ph...        1  
3  https://i.gr-assets.com/images/S/compressed.ph...        1  
4  https://i.gr-assets.com/images/S/compressed.ph...        1